tls/s2n_client_cert_request.c

```diff
@@ -138,7 +138,7 @@ int s2n_client_cert_req_send(struct s2n_connection *conn)
         GUARD(s2n_stuffer_write_uint8(out, s2n_cert_type_preference_list[i]));
```

Comment:
client cert request now has a context field, similar to Certificate for TLS 1.3

ref: https://tools.ietf.org/html/rfc8446#section-4.3.2

```
struct {
          opaque certificate_request_context<0..2^8-1>;
          Extension extensions<2..2^16-1>;
      } CertificateRequest;
```

here's an output from wireshark if it helps

```
TLSv1.3 Record Layer: Handshake Protocol: Certificate Request
    Opaque Type: Application Data (23)
    Version: TLS 1.2 (0x0303)
    Length: 171
    [Content Type: Handshake (22)]
    Handshake Protocol: Certificate Request
        Handshake Type: Certificate Request (13)
        Length: 150
        Certificate Request Context Length: 0
        Extensions Length: 147
        Extension: signature_algorithms (len=38)
            Type: signature_algorithms (13)
            Length: 38
            Signature Hash Algorithms Length: 36
            Signature Hash Algorithms (18 algorithms)
                Signature Algorithm: ecdsa_secp256r1_sha256 (0x0403)
                Signature Algorithm: ecdsa_secp384r1_sha384 (0x0503)
                Signature Algorithm: ecdsa_secp521r1_sha512 (0x0603)
                Signature Algorithm: ed25519 (0x0807)
                Signature Algorithm: ed448 (0x0808)
                Signature Algorithm: rsa_pss_pss_sha256 (0x0809)
                Signature Algorithm: rsa_pss_pss_sha384 (0x080a)
                Signature Algorithm: rsa_pss_pss_sha512 (0x080b)
                Signature Algorithm: rsa_pss_rsae_sha256 (0x0804)
                Signature Algorithm: rsa_pss_rsae_sha384 (0x0805)
                Signature Algorithm: rsa_pss_rsae_sha512 (0x0806)
                Signature Algorithm: rsa_pkcs1_sha256 (0x0401)
                Signature Algorithm: rsa_pkcs1_sha384 (0x0501)
                Signature Algorithm: rsa_pkcs1_sha512 (0x0601)
                Signature Algorithm: SHA224 ECDSA (0x0303)
                Signature Algorithm: ecdsa_sha1 (0x0203)
                Signature Algorithm: SHA224 RSA (0x0301)
                Signature Algorithm: rsa_pkcs1_sha1 (0x0201)
        Extension: certificate_authorities (len=101)
            Type: certificate_authorities (47)
            Length: 101
            Distinguished Names Length: 99
            Distinguished Names (99 bytes)
                Distinguished Name Length: 97
                Distinguished Name: (id-at-commonName=localhost,id-at-organizationalUnitName=s2n,id-at-organizationName=Amazon,id-at-localityName=Seattle,id-at-stateOrProvinceName=WA,id-at-countryName=US)
                    RDNSequence item: 1 item (id-at-countryName=US)
                    RDNSequence item: 1 item (id-at-stateOrProvinceName=WA)
                    RDNSequence item: 1 item (id-at-localityName=Seattle)
                    RDNSequence item: 1 item (id-at-organizationName=Amazon)
                    RDNSequence item: 1 item (id-at-organizationalUnitName=s2n)
                    RDNSequence item: 1 item (id-at-commonName=localhost)
```

Reply:
Do we also have to tweak the lengths we're sending? This has an overall extensions length and 1.2 only sends the length of signature_algorithms and certificate_authorities individually. We'll have to change `s2n_client_cert_req_recv` as well